### ISSUE TITLE
`solid/reactivity` inconsistently handles signals and memos when passed to functions

### ISSUE THREAD

**Issue by Huliiiiii** [solidjs-community/eslint-plugin-solid#182]
```tsx
const createCounter = (incrValue: () => number) => {
    const [count, setCount] = createSignal(0)
    return {
      count,
      incr: () => setCount((v) => v + incrValue()),
    }
  }

const [incr, setIncr] = createSignal(1)

const memo = createMemo(() => incr())

const counter = createCounter(incr) // Ok
const counter = createCounter(memo) // Error: The reactive variable 'memo' should be used within JSX, a tracked scope (like createEffect), or inside an event handler function, or else changes will be ignored.
```

Passing memo to functions like `createCounter` should not cause errors.

**Comment by ryansolid** [comment]
Confirmed on current main: `createCounter(incr)` passes while `createCounter(memo)` warns — signals and memos should be symmetrical when passed to `create*`/`use*` functions. Grouped with the 0.16.1 reactivity precision pass.

**Comment by ryansolid** [comment]
Fixed in v0.16.1. Memo accessors passed to `create*`/`use*`/custom reactive functions are now treated the same as signals — the rule resolves the variable to its `createMemo` call and applies the same allowance.
